### PR TITLE
[faiss 1.7.1] Impl RefineFlatIndex

### LINKED ISSUE #18: Add support for IndexRefineFlat
Hello.

faiss-sys supported IndexRefineFlat functions.

### PR DESCRIPTION
**Pull request by ava57r**
fixes #18 

https://github.com/facebookresearch/faiss/pull/1821

### CODE REVIEW

**Comment by Enet4** [comment]
We'll hopefully be able to review this PR and the other ones once facebookresearch/faiss#1915 is resolved.

**Comment by ava57r** [comment]
faiss release tagged [v1.7.1](https://github.com/facebookresearch/faiss/releases/tag/v1.7.1)

File: src/index/refine_flat.rs [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+use std::ptr;
+
+/// Alias for the native implementation of a index.
+pub type RefineFlatIndex = RefineFlatIndexImpl;
+
+/// Native implementation of a RefineFlat index.
+#[derive(Debug)]
+pub struct RefineFlatIndexImpl {
+    inner: *mut FaissIndexRefineFlat,
+}
```
Comment (Enet4) [inline src/index/refine_flat.rs:18]: We will need to keep track of the underlying index implementation and lifetime, and make the necessary changes accordingly.
```suggestion
pub struct RefineFlatIndexImpl<'a, I> {
    inner: *mut FaissIndexRefineFlat,
    phantom: std::marker::PhantomData<&'a I>,
}
```

**Review by Enet4 — CHANGES_REQUESTED** [review]
This one appears to have the same issues reported in #24. Please look into the pre-transform index and apply the same pattern of resolution here.

**Review by Enet4 — APPROVED** [review]
Looks good now, thank you!

**Pull request merged**